File: tutorials/golang/09-hello-liveness-readiness-health/main.go
package main

import (
	"fmt"
	"net/http"
)

func main() {
	dt := NewDependencyTracker()
	dt.Register("database", true, "connected")
	dt.Register("cache", true, "connected")

	mux := newHealthMux(dt)
	addr := ":4063"
	fmt.Printf("Listening on %s\n", addr)
	fmt.Println("  GET /healthz — liveness probe")
	fmt.Println("  GET /readyz  — readiness probe")
	fmt.Println("  GET /health  — detailed health")
	if err := http.ListenAndServe(addr, mux); err != nil {
		fmt.Printf("Error: %v\n", err)
	}
}

File: tutorials/rust/09-hello-liveness-readiness-health/src/main.rs
use hello_liveness_readiness_health::{app, AppState};
use tokio::net::TcpListener;

#[tokio::main]
async fn main() {
    let state = AppState::new(vec![("database", true), ("cache", true)]);
    let listener = TcpListener::bind("0.0.0.0:3000").await.unwrap();
    println!("Listening on http://0.0.0.0:3000");
    println!("  GET /healthz - liveness");
    println!("  GET /readyz  - readiness");
    println!("  GET /health  - detailed health");
    axum::serve(listener, app(state)).await.unwrap();
}
